src/platform/silabs/wifi/SiWx/ncp/efx32_ncp_host.c

```diff
@@ -95,47 +98,11 @@ static void efx32_spi_init(void)
     NVIC_SetPriority(GPIO_ODD_IRQn, PACKET_PENDING_INT_PRI);
     GPIOINT_CallbackRegister(INTERRUPT_PIN.pin, gpio_interrupt);
     GPIO_PinModeSet(INTERRUPT_PIN.port, INTERRUPT_PIN.pin, gpioModeInputPullFilter, 0);
-    GPIO_ExtIntConfig(INTERRUPT_PIN.port, INTERRUPT_PIN.pin, INTERRUPT_PIN.pin, true, false, true);
-}
 
-Ecode_t si91x_SPIDRV_MTransfer(SPIDRV_Handle_t handle, const void * txBuffer, void * rxBuffer, int count,

```

Comment:
this function was both sync and async in nature, sync when the payload was sub 16 bytes and async otherwise. The sub 16 bytes transfer logic was only to gain more throughput, additionally it is causing init failures in MATTER when SPI is involved in multiple slave communication. Hence dropping the logic and streamlining everything from DMA. This does typically mean a drop of 1-2mbps as seen by wiseconnect in their testing.